### PR TITLE
AGENT-203: Add ClusterImageSet asset

### PR DESCRIPTION
**Pull request by rwsu**
The ClusterImageSet is a Hive CRD used to specify which version
of OpenShift to install.

### CODE REVIEW

File: pkg/asset/agent/manifests/clusterimageset.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+// Files returns the files generated by the asset.
+func (a *ClusterImageSet) Files() []*asset.File {
+	if a.File != nil {
+		return []*asset.File{a.File}
+	}
+	return []*asset.File{}
+}
+
+// Load returns agentclusterinstall asset from the disk.
```
Comment (bfournie) [inline pkg/asset/agent/manifests/clusterimageset.go:50]: Nit - change agentclusterinstall

File: pkg/asset/agent/manifests/clusterimageset.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	clusterImageSetFilename = filepath.Join(clusterManifestDir, "cluster-image-set.yaml")
+)
+
+// ClusterImageSet generates the agent-cluster-install.yaml file.
```
Comment (bfournie) [inline pkg/asset/agent/manifests/clusterimageset.go:18]: Nit - "agent-cluster-install.yaml"

**Comment by bfournie** [comment]
It looks like a failure in ci/prow/verify-vendor due to:
github.com/openshift/hive/apis v0.0.0-20210506000654-5c038fb05190 // indirect

I think the *indirect* needs to be removed like Zane did in https://github.com/openshift/installer/pull/5933 since its now explicitly imported.

File: pkg/asset/agent/manifests/clusterimageset.go (hunk 1 above)
Comment (rwsu) [reply pkg/asset/agent/manifests/clusterimageset.go:50]: Thanks for spotting these.

**Pull request merged**